fe/fe-core/src/main/java/com/starrocks/common/util/AutoInferUtil.java

```diff
@@ -21,14 +21,14 @@
 
 public class AutoInferUtil {
     public static int calDefaultReplicationNum() throws UserException {
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
```

Comment:
should add simple UT to cover this utility. should be fairly easy to discover the issue at the first place.